services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1145,13 +1146,36 @@ Optional<String> getTopicForCurrentPushJob(
       latestTopic = Optional.of(versionTopics.get(0));
     }
 
+    Map<String, String> futureVersions = getFutureVersionsForMultiColos(clusterName, storeName);
+    for (Map.Entry<String, String> entry: futureVersions.entrySet()) {
+      ControllerClient controllerClient = getVeniceHelixAdmin().getControllerClientMap(clusterName).get(entry.getKey());
+      StoreResponse storeResponse = controllerClient.retryableRequest(5, c -> c.getStore(storeName));
```

Comment:
This was the two queries scenario I was trying to avoid; if "futureVersions" returned from call "getFutureVersionsForMultiColos" is not empty, why can't we just abort immediately?

Reply:
Discussed offline:
During target colo push, there is a phase where the target colo finishes the first push and no future version, while the rest of the colos haven't recieved the new push yet; at this moment, there is no future version in any child colos, and thus it's blocker for removing parent VT until we have a new state in parent controller to indicate that a push (either a regular push or single-colo push is ongoing.)